Unable to build for Android with versions 1.2.0-alpha01-devXX

I'm currently unable to built for Android using any of the dev builds for `1.2.0-alpha01-devXX`, although building for desktop works great.


The issue seems to be in the resolution of some android dependencies with a `-debug` suffix:


```
FAILURE: Build completed with 8 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':android:mergeDebugResources'.
> Could not resolve all files for configuration ':android:debugRuntimeClasspath'.
   > Could not find org.jetbrains.compose.foundation:foundation-layout-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.foundation:foundation-layout:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.material:material-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.material:material:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.animation:animation-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.animation:animation:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.runtime:runtime-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.runtime:runtime:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.ui:ui-util-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.ui:ui-util:1.2.0-alpha01-dev606

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':android:checkDebugAarMetadata'.
> Could not resolve all files for configuration ':android:debugRuntimeClasspath'.
   > Could not find org.jetbrains.compose.foundation:foundation-layout-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.foundation:foundation-layout:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.material:material-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.material:material:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.animation:animation-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.animation:animation:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.runtime:runtime-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.runtime:runtime:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.ui:ui-util-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.ui:ui-util:1.2.0-alpha01-dev606

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

3: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':android:mergeDebugAssets'.
> Could not resolve all files for configuration ':android:debugRuntimeClasspath'.
   > Could not find org.jetbrains.compose.foundation:foundation-layout-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.foundation:foundation-layout:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.material:material-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.material:material:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.animation:animation-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.animation:animation:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.runtime:runtime-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.runtime:runtime:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.ui:ui-util-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.ui:ui-util:1.2.0-alpha01-dev606

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

4: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':android:processDebugMainManifest'.
> Could not resolve all files for configuration ':android:debugRuntimeClasspath'.
   > Could not find org.jetbrains.compose.foundation:foundation-layout-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.foundation:foundation-layout:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.material:material-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.material:material:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.animation:animation-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.animation:animation:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.runtime:runtime-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.runtime:runtime:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.ui:ui-util-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.ui:ui-util:1.2.0-alpha01-dev606

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

5: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':common:generateDebugRFile'.
> Could not resolve all files for configuration ':common:debugCompileClasspath'.
   > Could not find org.jetbrains.compose.ui:ui-util-android-debug:1.2.0-alpha01-dev606.
     Required by:
         project :common > org.jetbrains.compose.ui:ui-util:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.runtime:runtime-android-debug:1.2.0-alpha01-dev606.
     Required by:
         project :common > org.jetbrains.compose.runtime:runtime:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.animation:animation-android-debug:1.2.0-alpha01-dev606.
     Required by:
         project :common > org.jetbrains.compose.animation:animation:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.material:material-android-debug:1.2.0-alpha01-dev606.
     Required by:
         project :common > org.jetbrains.compose.material:material:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.foundation:foundation-layout-android-debug:1.2.0-alpha01-dev606.
     Required by:
         project :common > org.jetbrains.compose.foundation:foundation-layout:1.2.0-alpha01-dev606

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

6: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':android:checkDebugDuplicateClasses'.
> Could not resolve all files for configuration ':android:debugRuntimeClasspath'.
   > Could not find org.jetbrains.compose.foundation:foundation-layout-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.foundation:foundation-layout:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.material:material-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.material:material:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.animation:animation-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.animation:animation:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.runtime:runtime-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.runtime:runtime:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.ui:ui-util-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.ui:ui-util:1.2.0-alpha01-dev606

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

7: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':android:desugarDebugFileDependencies'.
> Could not resolve all files for configuration ':android:debugRuntimeClasspath'.
   > Could not find org.jetbrains.compose.foundation:foundation-layout-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.foundation:foundation-layout:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.material:material-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.material:material:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.animation:animation-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.animation:animation:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.runtime:runtime-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.runtime:runtime:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.ui:ui-util-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.ui:ui-util:1.2.0-alpha01-dev606

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

8: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':android:mergeDebugNativeLibs'.
> Could not resolve all files for configuration ':android:debugRuntimeClasspath'.
   > Could not find org.jetbrains.compose.foundation:foundation-layout-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/foundation/foundation-layout-android-debug/1.2.0-alpha01-dev606/foundation-layout-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.foundation:foundation-layout:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.material:material-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/material/material-android-debug/1.2.0-alpha01-dev606/material-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.material:material:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.animation:animation-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/animation/animation-android-debug/1.2.0-alpha01-dev606/animation-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.animation:animation:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.runtime:runtime-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/runtime/runtime-android-debug/1.2.0-alpha01-dev606/runtime-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.runtime:runtime:1.2.0-alpha01-dev606
   > Could not find org.jetbrains.compose.ui:ui-util-android-debug:1.2.0-alpha01-dev606.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
       - https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/ui/ui-util-android-debug/1.2.0-alpha01-dev606/ui-util-android-debug-1.2.0-alpha01-dev606.pom
     Required by:
         project :android > org.jetbrains.compose.ui:ui-util:1.2.0-alpha01-dev606

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/7.0.2/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 5s
46 actionable tasks: 41 executed, 5 up-to-date


```

--- a/buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeImplPlugin.kt
+++ b/buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeImplPlugin.kt
@@ -90,6 +90,7 @@ class AndroidXComposeImplPlugin : Plugin<Project> {
 
                     if (plugin is KotlinMultiplatformPluginWrapper) {
                         project.configureForMultiplatform()
+                        OELPublishingPrototype(project)
                     }
                 }
             }

--- a/buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeImplPlugin.kt
+++ b/buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeImplPlugin.kt
@@ -90,7 +90,7 @@ class AndroidXComposeImplPlugin : Plugin<Project> {
 
                     if (plugin is KotlinMultiplatformPluginWrapper) {
                         project.configureForMultiplatform()
-                        OELPublishingPrototype(project)
+                        enableOELPublishing(project)
                     }
                 }
             }

--- a/buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeMultiplatformExtensionImpl.kt
+++ b/buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeMultiplatformExtensionImpl.kt
@@ -21,6 +21,16 @@ import org.jetbrains.kotlin.gradle.plugin.KotlinJsCompilerType
 import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSet
 import org.gradle.api.Project
 import javax.inject.Inject
+import org.jetbrains.kotlin.gradle.plugin.*
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinAndroidTarget
+import kotlin.reflect.full.memberProperties
+import org.jetbrains.kotlin.gradle.plugin.mpp.*
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.internal.dependencies.DefaultMavenDependency
+import org.gradle.api.publish.maven.internal.dependencies.MavenDependencyInternal
+import org.gradle.api.publish.maven.internal.publication.DefaultMavenPublication
+import org.gradle.api.attributes.Usage
+
 
 open class AndroidXComposeMultiplatformExtensionImpl @Inject constructor(
     project: Project
@@ -121,3 +131,93 @@ open class AndroidXComposeMultiplatformExtensionImpl @Inject constructor(
             }
     }
 }
+
+fun Project.experimentalOELPublication() : Boolean = findProperty("oel.publication") == "true"
+fun Project.oelAndroidxVersion() : String? = findProperty("oel.androidx.version") as String?
+
+
+fun OELPublishingPrototype(project: Project) {
+    if (!project.experimentalOELPublication()) return
+
+    if (project.experimentalOELPublication() && (project.oelAndroidxVersion() == null)) {
+        error("androidx version should be specified for OEL publications")
+    }
+
+
+    val ext = project.multiplatformExtension ?: error("expected a multiplatform project")
+
+    ext.targets.all { target ->
+        if (target is KotlinAndroidTarget) {
+            project.publishAndroidxReference(target)
+        }
+    }
+}
+
+// FIXME: reflection access! Some API in Kotlin is needed
+@Suppress("unchecked_cast")
+private val KotlinTarget.kotlinComponents: Iterable<KotlinTargetComponent>
+    get() = javaClass.kotlin.memberProperties
+        .single { it.name == "kotlinComponents" }
+        .get(this) as Iterable<KotlinTargetComponent>
+
+
+@Suppress("unchecked_cast")
+private fun Project.publishAndroidxReference(target: KotlinTarget) {
+    afterEvaluate {
+        target.kotlinComponents.forEach { component ->
+            val componentName = component.name
+
+            val multiplatformExtension =
+                extensions.findByType(KotlinMultiplatformExtension::class.java)
+                    ?: error("Expected a multiplatform project")
+
+            if (component is KotlinVariant)
+                component.publishable = false
+
+            val usages = when (component) {
+                is KotlinVariant -> component.usages
+                is JointAndroidKotlinTargetComponent -> component.usages
+                else -> emptyList()
+            }
+
+            extensions.getByType(PublishingExtension::class.java)
+                .publications.withType(DefaultMavenPublication::class.java)
+                // isAlias is needed for Gradle to ignore the fact that there's a
+                // publication that is not referenced as an available-at variant of the root module
+                // and has the Maven coordinates that are different from those of the root module
+                // FIXME: internal Gradle API! We would rather not create the publications,
+                //        but some API for that is needed in the Kotlin Gradle plugin
+                .all { publication ->
+                    if (publication.name == componentName) {
+                        publication.isAlias = true
+                    }
+                }
+
+            usages.forEach {    usage ->
+                val configurationName = usage.name + "-published"
+
+                configurations.matching{it.name == configurationName}.all() { conf ->
+                    conf.artifacts.clear()
+                    conf.dependencies.clear()
+                    conf.setExtendsFrom(emptyList())
+                    var version = if (target.project.group.toString().contains("org.jetbrains.compose.material3")) "1.0.0-alpha01" else target.project.oelAndroidxVersion()!!
+                    val newDependency = target.project.group.toString().replace("org.jetbrains.compose", "androidx.compose") + ":" + name + ":" + version
+                    conf.dependencies.add(target.project.dependencies.create(newDependency))
+                }
+
+                val rootComponent : KotlinSoftwareComponent = target.project.components.withType(KotlinSoftwareComponent::class.java)
+                    .getByName("kotlin")
+
+                (rootComponent.usages as MutableSet).add(
+                    DefaultKotlinUsageContext(
+                        multiplatformExtension.metadata().compilations.getByName("main"),
+                        objects.named(Usage::class.java, "kotlin-api"),
+                        configurationName
+                    )
+                )
+
+            }
+        }
+    }
+}
+

--- a/buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeMultiplatformExtensionImpl.kt
+++ b/buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeMultiplatformExtensionImpl.kt
@@ -134,9 +134,10 @@ open class AndroidXComposeMultiplatformExtensionImpl @Inject constructor(
 
 fun Project.experimentalOELPublication() : Boolean = findProperty("oel.publication") == "true"
 fun Project.oelAndroidxVersion() : String? = findProperty("oel.androidx.version") as String?
+fun Project.oelAndroidxMaterial3Version() : String? = findProperty("oel.androidx.material3.version") as String?
 
 
-fun OELPublishingPrototype(project: Project) {
+fun enableOELPublishing(project: Project) {
     if (!project.experimentalOELPublication()) return
 
     if (project.experimentalOELPublication() && (project.oelAndroidxVersion() == null)) {
@@ -200,7 +201,14 @@ private fun Project.publishAndroidxReference(target: KotlinTarget) {
                     conf.artifacts.clear()
                     conf.dependencies.clear()
                     conf.setExtendsFrom(emptyList())
-                    var version = if (target.project.group.toString().contains("org.jetbrains.compose.material3")) "1.0.0-alpha01" else target.project.oelAndroidxVersion()!!
+                    val composeVersion = requireNotNull(target.project.oelAndroidxVersion()) {
+                        "Please specify oel.androidx.version property"
+                    }
+                    val material3Version = requireNotNull(target.project.oelAndroidxMaterial3Version()) {
+                        "Please specify oel.androidx.material3.version property"
+                    }
+
+                    val version = if (target.project.group.toString().contains("org.jetbrains.compose.material3")) material3Version else composeVersion
                     val newDependency = target.project.group.toString().replace("org.jetbrains.compose", "androidx.compose") + ":" + name + ":" + version
                     conf.dependencies.add(target.project.dependencies.create(newDependency))
                 }

